doc/reference/trd-dynamic-process-loading.md

```diff
@@ -195,6 +212,21 @@ Implementations may include a policy for choosing whether to load a new process
 binary. The implementation must also use the board's chosen credential checking
 policy.
 
+---------------------------------
+
+The methods described for both the traits return a synchronous result to the 
+userland app. If the result is `Ok(())`, the operation requested operation can 
+be performed and the the userland app must wait for an asynchronous upcall 
+indicating the result of the operation. 
+If the result is a variant of `ErrorCode`, the kernel deallocates the 
+resources reserved for the new application and writes a padding application 
+in its stead. The userland application must request for resources with the 
+`setup()` command and restart the process once again. This method reduces 
+potential fragmentation due to problematic binaries.
```

Comment:
This can maybe move lower, but it is too specific (e.g. what is a padding app?) for this section of the TRD.